lib/connection/connection.js

```diff
@@ -205,6 +209,47 @@ var closeHandler = function(self) {
   }
 }
 
+var decompress = function(compressorId, compressedData, callback) {
+  var decompressors = [
+    function(compressedData, callback) {
+      callback(null, compressedData)
+    },
+    Snappy.uncompress,
+    zlib.unzip
+  ];
+
+  if (compressorId < 0 || compressorId > decompressors.length - 1) {
+    throw new Error('Server sent message compressed using an unsupported compressor');
```

Comment:
Include the compressor id in the message to help the user identify what's gone wrong